lib/puppet/application/master.rb

```diff
@@ -256,16 +249,6 @@ def setup_ssl
     Puppet::SSL::Oids.load_custom_oid_file(Puppet[:trusted_oid_mapping_file])
   end
 
-  # Sets up a special node cache "write only yaml" that collects and stores node data in yaml
-  # but never finds or reads anything (this since a real cache causes stale data to be served
-  # in circumstances when the cache can not be cleared).
-  # @see puppet issue 16753
-  # @see Puppet::Node::WriteOnlyYaml
-  # @return [void]
-  def setup_node_cache
-    Puppet::Node.indirection.cache_class = Puppet[:node_cache_terminus]
-  end
-
```

Comment:
Discussing with @joshcooper we're going to leave this method in to honor the setting value in the master application to enable a user to keep the value if they'd like to. That should be more discoverable/easier than routes.yaml.